apps/vercel/frontend/src/clients/Vercel.ts

```diff
@@ -37,6 +44,44 @@ export default class VercelClient implements VercelAPIClient {
     return data;
   }
 
+  async listDeploymentSummaries(
+    projectId: string,
+    deploymentId?: string
+  ): Promise<ListDeploymentSummaryResponse> {
+    const latestDeploymentId = deploymentId || (await this.getLatestDeploymentId(projectId));
+    const res = await fetch(
+      `${this.baseEndpoint}/v6/deployments/${latestDeploymentId}/files/outputs?file=..%2Fdeploy_metadata.json`,
+      {
+        headers: this.buildHeaders(),
+        method: 'GET',
+      }
+    );
+
+    const data = await res.json();
+
+    return data;
+  }
+
+  async listApiPaths(projectId: string): Promise<ApiPath[]> {
+    const data = await this.listDeploymentSummaries(projectId);
+
+    const apiPaths = this.filterServerlessFunctions(data.serverlessFunctions);
+    const formattedApiPaths = this.formatApiPaths(apiPaths);
+
+    return formattedApiPaths;
+  }
+
+  async getLatestDeploymentId(projectId: string): Promise<string> {
+    const res = await fetch(`${this.baseEndpoint}/v6/deployments?projectId=${projectId}`, {
+      headers: this.buildHeaders(),
+      method: 'GET',
+    });
+
+    const data = await res.json();
+
+    return data.deployments[0].uid;
```

Comment:
Have added a ticket to further investigate if this is the "way" to grab the latest deploymentId, or if we can more accurately target this.